includes/Maps_DisplayMapRenderer.php

```diff
@@ -225,28 +229,35 @@ public static function evilOpenLayersHack( $layers ) {
 				}
 			}
 		}
-
-		MapsMappingServices::getServiceInstance( 'openlayers' )->addLayerDependencies( self::getLayerDependencies( $layerNames ) );
-
 		return $layerDefs;
 	}
 
-	/**
-	 * FIXME
-	 * @see evilOpenLayersHack
-	 */
-	private static function getLayerDependencies( array $layerNames ) {
-		global $egMapsOLLayerDependencies, $egMapsOLAvailableLayers;
+	public static function getLayerDependencies( $service, $params ) {
```

Comment:
Yes its used in `evilOpenLayersHack()` with was used in Semantic Maps. I did a change in https://github.com/SemanticMediaWiki/SemanticMaps/pull/107. So this change destroys some backward compatibility. Any Ideas how to prevent this?